vlc4: init at 4.0.0-unstable-2024-12-06

PerchunPak opened the pull request:
This is really far from perfect, but at least it builds. I am not sure if v4 so WIP or I packaged it badly

Would be great if someone would help to iterate on this

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Fixes #305581

CC @alois31

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

PerchunPak committed:
vlc4: init at 4.0.0-unstable-2024-12-29

PerchunPak commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 362352`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vlc4</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>vlc4</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

alois31 commented:
Thank you for your work on this, it is much appreciated!

> This is really far from perfect, but at least it builds. I am not sure if v4 so WIP or I packaged it badly

Unfortunately it seems to be a combination of both. I have also tried building VLC 4 on a more traditional distro (in a VM), and while it is a bit weird, it can at least play videos. On NixOS (on bare metal) it just crashes with the following error:
```
[00007dd60a5fe628] xcb generic error: window not available
wp_viewporter#58: error 0: the specified surface already has a viewport
```

I should really give this a look at some point, there are also other nixpkgs-only issues in VLC 3 already that seem to be related to incorrect interaction with external libraries.

A small nit at last: 4.0.0-unstable-* is not really following the version convention here, since there is no 4.0.0 release. It is probably debatable whether it should be 3.0.0-unstable-* or 0-unstable-*.

PerchunPak commented:
> ```
> [00007dd60a5fe628] xcb generic error: window not available
> wp_viewporter#58: error 0: the specified surface already has a viewport
> ```

Weird, I have entirely different error

```
VLC media player 4.0.0-dev Otto Chriek (revision c7bac983d56d8c9a786791603657567372c262f8)
[000000001f401830] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
libva info: VA-API version 1.22.0
libva info: User environment variable requested driver 'nvidia'
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0
[000076db2d128d00] vaapi generic: Using VA-API NVDEC driver [direct backend]
[000076db2d12e310] glinterop_vaapi gl error: vaDeriveImage: operation failed
[000076db2d12e310] gl gl error: Could not create interop
[000076db2d130b30] glinterop_vaapi gl error: vaDeriveImage: operation failed
[000076db2d130b30] gles2 gl error: Could not create interop
[000076db7f7f48f8] xcb generic error: window not available
[000076db08000bb0] blend blend error: no matching alpha blending routine (chroma: YUVA -> VAOP)
[000076db08000bb0] main blend error: blending YUVA to VAOP failed
[000076db2d138f40] vaapi_filters filter error: vaDeriveImage: operation failed
[000076db08000bb0] blend blend error: no matching alpha blending routine (chroma: YUVA -> VAOP)
[000076db08000bb0] main blend error: blending YUVA to VAOP failed
[000076db08000bb0] blend blend error: no matching alpha blending routine (chroma: YUVA -> VAOP)
[000076db08000bb0] main blend error: blending YUVA to VAOP failed
[000076db08000bb0] blend blend error: no matching alpha blending routine (chroma: YUVA -> VAOP)
[000076db08000bb0] main blend error: blending YUVA to VAOP failed
[000076db08000bb0] blend blend error: no matching alpha blending routine (chroma: YUVA -> VAOP)
[000076db08000bb0] main blend error: blending YUVA to VAOP failed
[000076db08000bb0] blend blend error: no matching alpha blending routine (chroma: YUVA -> VAOP)
[000076db08000bb0] main blend error: blending YUVA to VAOP failed
```

What's your hardware? Mine: CPU AMD Ryzen 7 5800X, NVIDIA GeForce RTX 3070 (drivers 565.77), Kernel 6.12.1-zen1

(looks like my error is related to nvidia gpu only)

> A small nit at last: 4.0.0-unstable-* is not really following the version convention here, since there is no 4.0.0 release. It is probably debatable whether it should be 3.0.0-unstable-* or 0-unstable-*.

Yes, there is no tag for 4.0.0, but this is what upstream chooses for (e.g.) official nightly [snaps](https://snapcraft.io/vlc).

---

I also decided to compare build logs (precisely `./configure` logs) between [Ubuntu 24.04 git20241129](https://launchpadlibrarian.net/761052372/buildlog_ubuntu-noble-amd64.vlc_4.0.0~rc1~~git20241129+r104884+388~ubuntu24.04.1_BUILDING.txt.gz) logs and this build and [found](https://github.com/NixOS/nixpkgs/compare/716640a1c7c26aa0fbd84ae6c3d9afe6bfab6c0b..0e3d0ee72866ad8308c0a913997eb5e0bfd49f53) some libraries that we miss (though it didn't help for me). I am unsure if VLC really needs those libraries, but it increased derivation size by 0.5gb

alois31 commented:
> Weird, I have entirely different error
> What's your hardware? Mine: CPU AMD Ryzen 7 5800X, NVIDIA GeForce RTX 3070 (drivers 565.77), Kernel 6.12.1-zen1
> 
> (looks like my error is related to nvidia gpu only)

AMD Ryzen 9 5900HX, iGPU only, kernel 6.12.1.

> Yes, there is no tag for 4.0.0, but this is what upstream chooses for (e.g.) official nightly [snaps](https://snapcraft.io/vlc).

Strictly speaking they use 4.0.0-dev, but it probably doesn't matter that much anyway.

> I also decided to compare build logs (precisely `./configure` logs) between [Ubuntu 24.04 git20241129](https://launchpadlibrarian.net/761052372/buildlog_ubuntu-noble-amd64.vlc_4.0.0~rc1~~git20241129+r104884+388~ubuntu24.04.1_BUILDING.txt.gz) logs and this build and [found](https://github.com/NixOS/nixpkgs/compare/716640a1c7c26aa0fbd84ae6c3d9afe6bfab6c0b..0e3d0ee72866ad8308c0a913997eb5e0bfd49f53) some libraries that we miss (though it didn't help for me). I am unsure if VLC really needs those libraries, but it increased derivation size by 0.5gb

Oddly enough, with your latest version VLC no longer crashes trying to play videos on my machine, so at least some of them seem to be needed in some way. Thank you!

PerchunPak commented:
Updated to the latest commit and rebased onto master. Now I have entirely different error:

```
VLC media player 4.0.0-dev Otto Chriek (revision 1e70019a37b98df18f66eacd98b98dfc8528014d)
[00000000255f9b80] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
libva info: VA-API version 1.22.0
libva info: User environment variable requested driver 'nvidia'
libva info: Trying to open /run/opengl-driver/lib/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_0
libva info: va_openDriver() returns 0
[000071835d129670] vaapi generic: Using VA-API NVDEC driver [direct backend]
wl_display#1: error 1: invalid arguments for wl_subcompositor#35.get_subsurface
[qt] (qt.qpa.wayland) eglSwapBuffers failed with 0x3000, surface: 0x7183cf8b75d0
```

Will open an upstream issue about this

Ramblurr commented:
I updated the pkg with:

```diff
diff --git a/pkgs/by-name/vl/vlc4/package.nix b/pkgs/by-name/vl/vlc4/package.nix
index 03957ff0a9d5..7a547be506ea 100644
--- a/pkgs/by-name/vl/vlc4/package.nix
+++ b/pkgs/by-name/vl/vlc4/package.nix
@@ -120,14 +120,14 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "vlc4";
-  version = "4.0.0-unstable-2024-12-29";
+  version = "4.0.0-unstable-2025-07-30";

   src = fetchFromGitLab {
     owner = "videolan";
     repo = "vlc";
-    rev = "dc430c6528d805ad57b6c2b1bd6a746ef2de12a8";
+    rev = "7c5cd6e7361fac223232b5d949c6cf27945c219e";
     domain = "code.videolan.org";
-    hash = "sha256-5RTQrWaJHL+whq+WOuW5+9XHrF6jayjWJtKusPNNnhU=";
+    hash = "sha256-+Stm2/VacgXOlux48Wi2ss/OPQjhU0odQTJ6XbGIRyw=";
   };

```

but when I run it it (on wayland fwiw) I get:

```
❯ ./result/bin/vlc
VLC media player 4.0.0-dev Otto Chriek (revision 7c5cd6e7361fac223232b5d949c6cf27945c219e)
[000000000116daf0] main libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
[00000000013eeef0] [qt] qt generic error: qml loading No such file or directory qrc:/qt/qml/VLC/MainInterface/MainInterface.qml:4294967295
[00000000013eeef0] [qt] qt generic error: Install missing modules using your packaging tool
[00000000013eeef0] [qt] qt generic: not running an X11 platform
[00000000013eeef0] [qt] qt generic error: qml loading No such file or directory qrc:/qt/qml/VLC/MainInterface/MainInterface.qml:4294967295
[00000000013eeef0] [qt] qt generic error: Install missing modules using your packaging tool
[00000000013eeef0] [qt] qt generic error: no suitable compositor found
[00000000013eeef0] [qt] qt generic error: unable to create main interface
[0000000001280e80] skins2 interface error: cannot instantiate dialogs provider
Remote control interface initialized. Type `help' for help.
```

PerchunPak commented:
Turns out mpv is enough...

If someone would like to continue my work, please feel free to do so. I lost interest in this package.
